### PR TITLE
fix: waitUntil when sending code/link email on universal auth to improve time to next page

### PR DESCRIPTION
**Pull request by danielgent**
As requested on https://github.com/sesamyab/auth/pull/870 - was easier to recreate the PR

We should manually A/B test this before and after merging

This is me running it on wrangler locally... I think it's marginally quicker. I prefer using machines to measure though :smile: 

[Screencast from 2024-06-03 11-41-16.webm](https://github.com/sesamyab/auth/assets/8496063/740fc4ac-7ab2-4e60-b75e-82e49d962a70)

### CODE REVIEW

File: src/routes/tsx/routes.tsx [hunk 1]
```diff
@@ -804,9 +805,12 @@ export const loginRoutes = new OpenAPIHono<{ Bindings: Env }>()
         magicLink.searchParams.set("verification_code", code);
         magicLink.searchParams.set("nonce", "nonce");
 
-        await sendLink(env, client, params.username, code, magicLink.href);
+        waitUntil(
+          ctx,
+          sendLink(env, client, params.username, code, magicLink.href),
+        );
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:811]: :eyes: no code changes anywhere else but here (so much safer than the other PR)

This is awaiting as high up the stack as possible

File: test/integration/login/code-flow-liquidjs.spec.ts [hunk 2]
```diff
@@ -114,6 +114,9 @@ describe("Login with code on liquidjs template", () => {
     expect(postSendCodeResponse.status).toBe(302);
     const enterCodeLocation = postSendCodeResponse.headers.get("location");
 
+    // flush pipe
+    await new Promise((resolve) => setTimeout(resolve, 0));
```
Comment (danielgent) [inline test/integration/login/code-flow-liquidjs.spec.ts:118]: a testing trick I haven't used for a while! will resolve all our `waitUntils` which are queued up next in the event loop

Comment (markusahlstrand) [reply test/integration/login/code-flow-liquidjs.spec.ts:118]: Think there's a waitForNextTick as well? But this should work fine :)

Comment (danielgent) [reply test/integration/login/code-flow-liquidjs.spec.ts:118]: Where is there what? :smile:

**Pull request merged**